### PR TITLE
CI implementation

### PR DESCRIPTION
**Pull request by MrYar0s**
[CI] Implementation
[Logger] Add simple logger
[Tests] Refactored tests

### CODE REVIEW

**Comment by MrYar0s** [comment]
NOTE: Rewriting parameters for vm now

File: assembler/gen_instr.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ def write_file_open(out: TextIOWrapper) :
 
             "// Get ptr to buff with instruction binary code\n"
             "virtual const Byte *getBinCode() const noexcept = 0;\n"
+
+            "// According to clang-tidy errors\n"
```
Comment (VasiliyMatr) [inline assembler/gen_instr.py]: Virtual destructor is actually needed. Remove comment?

File: common/include/shrimp/common/logger.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#ifndef SHRIMP_COMMON_LOGGER_HPP
+#define SHRIMP_COMMON_LOGGER_HPP
+
+#include <iostream>
+#include <cstdint>
+
+namespace shrimp {
+
+enum LogLevel : uint8_t { NONE = 0, ERROR, INFO, DEBUG };
```
Comment (VasiliyMatr) [inline common/include/shrimp/common/logger.hpp:9]: Enum class?

File: runtime/shrimp_vm/include/shrimp/runtime/shrimp_vm.hpp [hunk 3]
```diff
@@ -30,6 +32,11 @@ class ShrimpVM final {
         return code_.data();
     }
 
+    LogLevel getLogLevel()
```
Comment (VasiliyMatr) [inline runtime/shrimp_vm/include/shrimp/runtime/shrimp_vm.hpp]:
```suggestion
    [[nodiscard]] auto getLogLevel() const noexcept
```

File: runtime/shrimp_vm/include/shrimp/runtime/shrimp_vm.hpp [hunk 4]
```diff
@@ -52,6 +59,7 @@ class ShrimpVM final {
 
 private:
     Runtime *runtime_ = nullptr;
+    LogLevel log_level_;
```
Comment (VasiliyMatr) [inline runtime/shrimp_vm/include/shrimp/runtime/shrimp_vm.hpp]:
```suggestion
    LogLevel log_level_ = LogLevel::NONE;
```

File: scripts/run_clang_tidy.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import sys
+import os
+import re
+
+PROGRAM = "clang-tidy"
```
Comment (VasiliyMatr) [inline scripts/run_clang_tidy.py:5]: Maybe just write "clang-tidy"?

File: shrimp/src/shrimp.cpp [hunk 6]
```diff
@@ -37,11 +42,40 @@ std::vector<Byte> readFromFile(std::string file_name)
     return code;
 }
 
+LogLevel getLogLevel(int argc, char *argv[])
```
Comment (VasiliyMatr) [inline shrimp/src/shrimp.cpp]: Remove?

File: shrimp/src/shrimp.cpp [hunk 7]
```diff
@@ -7,8 +7,13 @@
...
+#include <shrimp/common/logger.hpp>
+
 #include <shrimp/runtime/shrimp_vm.hpp>
 
+#include <CLI/CLI.hpp>
+#include <CLI/App.hpp>
+
 namespace shrimp {
 
 std::string getFileName(int argc, char *argv[])
```
Comment (VasiliyMatr) [inline shrimp/src/shrimp.cpp]: Remove?

File: shrimp/src/shrimp.cpp [hunk 8]
```diff
@@ -37,11 +42,40 @@ std::vector<Byte> readFromFile(std::string file_name)
...
+    }
+    return getLogLevelByString(argv[pos + 1]);
+}
+
 int Main(int argc, char *argv[])
 {
+    CLI::App app("Shrimp VM");
+
+    std::string log_level_str {};
+    auto *log_level_cli = app.add_option("--log-level", log_level_str, "Log level file [error, info, debug, none]");
```
Comment (VasiliyMatr) [inline shrimp/src/shrimp.cpp]:
```suggestion
    auto *log_level_cli = app.add_option("--log-level", log_level_str, "Log level [error, info, debug, none]");
```

File: shrimp/src/shrimp.cpp [hunk 9]
```diff
@@ -37,11 +42,40 @@ std::vector<Byte> readFromFile(std::string file_name)
...
+    log_level_cli->default_str("none");
+
+    std::string input_file {};
+    auto *input_arg = app.add_option("--in", input_file, "Input file [Required as last parameter]");
```
Comment (VasiliyMatr) [inline shrimp/src/shrimp.cpp]:
```suggestion
    auto *input_arg = app.add_option("--in", input_file, "Input file");
```

File: shrimp/src/shrimp.cpp [hunk 10]
```diff
@@ -37,11 +42,40 @@ std::vector<Byte> readFromFile(std::string file_name)
...
+    input_arg->required();
+
+    CLI11_PARSE(app, argc, argv);
+
+    LogLevel log_level = getLogLevelByString(log_level_str);
     std::string file_name = getFileName(argc, argv);
     auto native_code = readFromFile(file_name);
```
Comment (VasiliyMatr) [inline shrimp/src/shrimp.cpp]:
```suggestion
    auto native_code = readFromFile(input_file);
```

File: shrimp/src/shrimp.cpp [hunk 11]
```diff
@@ -37,11 +42,40 @@ std::vector<Byte> readFromFile(std::string file_name)
...
+    log_level_cli->default_str("none");
+
+    std::string input_file {};
+    auto *input_arg = app.add_option("--in", input_file, "Input file [Required as last parameter]");
+    input_arg->required();
+
+    CLI11_PARSE(app, argc, argv);
+
+    LogLevel log_level = getLogLevelByString(log_level_str);
     std::string file_name = getFileName(argc, argv);
```
Comment (VasiliyMatr) [inline shrimp/src/shrimp.cpp]:
```suggestion
```

File: shrimp/src/shrimp.cpp [hunk 12]
```diff
@@ -37,11 +42,40 @@ std::vector<Byte> readFromFile(std::string file_name)
...
     auto native_code = readFromFile(file_name);
-    runtime::ShrimpVM svm {native_code};
+    runtime::ShrimpVM svm {native_code, log_level};
 
     return svm.runImpl();
```
Comment (VasiliyMatr) [inline shrimp/src/shrimp.cpp:52]: Maybe print status?

File: tests/cts_tests/interpreter/CMakeLists.txt [hunk 13]
```diff
@@ -6,14 +6,17 @@ function(shrimp_cts_test test_name)
 	set(TEST_SOURCE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${test_name}.shr)
 	file(MAKE_DIRECTORY ${TEST_BUILD_DIR})
 
-	add_custom_command(COMMAND cd ${TEST_BUILD_DIR} && ${PROJECT_BINARY_DIR}/assembler/assembler < ${TEST_SOURCE_PATH}
-		OUTPUT ${TEST_BUILD_DIR}/${test_name}.imp
+	add_custom_target(compile_cts_${test_name}
+		COMMAND cd ${TEST_BUILD_DIR} && ${PROJECT_BINARY_DIR}/assembler/assembler
+		--in ${TEST_SOURCE_PATH}
+		--out ${TEST_BUILD_DIR}/${test_name}.imp
 		DEPENDS assembler
```
Comment (VasiliyMatr) [inline tests/cts_tests/interpreter/CMakeLists.txt]:
```suggestion
		DEPENDS assembler ${TEST_SOURCE_PATH}
```

File: tests/e2e_tests/CMakeLists.txt [hunk 14]
```diff
@@ -6,14 +6,17 @@ function(shrimp_e2e_test test_name)
 	set(TEST_SOURCE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${test_name}.shr)
 	file(MAKE_DIRECTORY ${TEST_BUILD_DIR})
 
-	add_custom_command(COMMAND cd ${TEST_BUILD_DIR} && ${PROJECT_BINARY_DIR}/assembler/assembler < ${TEST_SOURCE_PATH}
-		OUTPUT ${TEST_BUILD_DIR}/${test_name}.imp
+	add_custom_target(compile_e2e_${test_name}
+		COMMAND cd ${TEST_BUILD_DIR} && ${PROJECT_BINARY_DIR}/assembler/assembler
+		--in ${TEST_SOURCE_PATH}
+		--out ${TEST_BUILD_DIR}/${test_name}.imp
 		DEPENDS assembler
```
Comment (VasiliyMatr) [inline tests/e2e_tests/CMakeLists.txt]:
```suggestion
		DEPENDS assembler ${TEST_SOURCE_PATH}
```

**Review by VasiliyMatr — CHANGES_REQUESTED** [review]
Looks fine. Consider my suggestions

**Pull request merged**